spec/prepending_initializer_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+RSpec.describe "prepending initializer" do # rubocop:disable RSpec/DescribeClass
+  context "when the class's initializer take arguments" do
+    context "when it only takes positional arguments" do
+      let(:class_with_memo) do
+        Class.new do
+          prepend MemoWise
+
+          def initialize(arg); end
+        end
+      end
+
+      it "does not raise an error when initializing the class" do
+        expect { class_with_memo.new(:pos) }.to_not raise_error
+      end
+    end
+
+    context "when it only takes keyword arguments" do
+      let(:class_with_memo) do
+        Class.new do
+          prepend MemoWise
+
+          def initialize(kwarg:); end
+        end
+      end
+
+      it "does not raise an error when initializing the class" do
+        expect { class_with_memo.new(kwarg: :kw) }.to_not raise_error
+      end
+    end
+
+    context "when it take both positional and keyword arguments" do
+      let(:class_with_memo) do
+        Class.new do
+          prepend MemoWise
+
+          def initialize(arg, kwarg:); end
+        end
+      end
+
+      it "does not raise an error when initializing the class" do
+        expect { class_with_memo.new(:pos, kwarg: :kw) }.to_not raise_error
+      end
+    end
+
+    context "when the method takes positional arguments, keyword arguments, "\
+              "and a block" do
+      let(:class_with_memo) do
+        Class.new do
+          prepend MemoWise
+
+          def initialize(arg, kwarg:, &block); end
```

Comment:
I don't know a ton about how Ruby handles block args but putting them in the params does not make them required:

```ruby
def foo(&block) ; end
foo
=> nil
```

and any method can have a block passed to it:

```ruby
def foo ; end
foo { true }
=> nil
```

Given this, would it make sense to add a `yield` to the `initialize` to make it more closely match real-world usage?

Reply:
I see your point! In this particular case, I see `block.call` as being more in keeping with the intent, how's that?

Reply:
Changed, is this good?